norlab_icp_mapper/MapperModules/ComputeDynamicsMapperModule.cpp

```diff
@@ -0,0 +1,166 @@
+#include <nabo/nabo.h>
+#include "ComputeDynamicsMapperModule.h"
+
+ComputeDynamicsMapperModule::ComputeDynamicsMapperModule(const PM::Parameters& params):
+	MapperModule("ComputeDynamicsMapperModule",ComputeDynamicsMapperModule::availableParameters(), params),
+	    priorDynamic(PM::Parametrizable::get<float>("priorDynamic")),
+	    thresholdDynamic(PM::Parametrizable::get<float>("thresholdDynamic")),
+	    alpha(PM::Parametrizable::get<float>("alpha")),
+	    beta(PM::Parametrizable::get<float>("beta")),
+	    beamHalfAngle(PM::Parametrizable::get<float>("beamHalfAngle")),
+	    epsilonA(PM::Parametrizable::get<float>("epsilonA")),
+		epsilonD(PM::Parametrizable::get<float>("epsilonD")),
+		sensorMaxRange(PM::Parametrizable::get<float>("sensorMaxRange")),
+		transformation(PM::get().TransformationRegistrar.create("RigidTransformation"))
+    {}
+
+PointMatcher<float>::DataPoints ComputeDynamicsMapperModule::createMap(const PM::DataPoints& input, const PM::TransformationParameters& pose)
+{
+    DataPoints outputMap(input);
+    inPlaceCreateMap(outputMap, pose);
+    return outputMap;
+}
+
+void ComputeDynamicsMapperModule::inPlaceCreateMap(PM::DataPoints& input, const PM::TransformationParameters& pose)
+{
+}
+
+PointMatcher<float>::DataPoints ComputeDynamicsMapperModule::updateMap(const PM::DataPoints& input, const PM::DataPoints& map, const PM::TransformationParameters& pose)
+{
+    DataPoints outputMap(map);
+    inPlaceUpdateMap(input, outputMap, pose);
+    return outputMap;
+}
+
+void ComputeDynamicsMapperModule::inPlaceUpdateMap(const PM::DataPoints& input, PM::DataPoints& map, const PM::TransformationParameters& pose)
+{
+    // assume the input already has the probabilityDynamic descriptor
+    // can be added in yaml config with the AddDescriptorDataPointsFilter
```

Comment:
The mapper module would be easier to use if it added the necessary descriptors on its own, is there any reason for not adding the descriptor here?

Reply:
It's an open question that I'm not sure about. The current design is more in the separation of concerns paradigm, giving the users more control and letting them deal with the consequences. I modified the exception message to make it explicitly clear how to fix the missing descriptor.

Reply:
So, after giving it more thought and playing with the code, I believe requiring the user to add the descriptor explicitly might still be the best option.
The updateMap method signature 
```C++
PM::DataPoints updateMap(const PM::DataPoints& input, const PM::DataPoints& map, const PM::TransformationParameters& pose)
```
takes a constant input point cloud. The method is called updateMap, so it makes sense that the input has not been modified. However, setting the descriptor in a `MapperModule` requires writing to the input variable. Alternatively, it is possible to make a copy and modify it. BUT the change does not propagate outside of the current `MapperModule`, so any subsequent gets called without this descriptor. The default libpointmatcher behavior on point cloud concatenation is that it removes the descriptors that are not present in both point clouds, which is exactly what happens when a PointDistance or Octree modules are caused afterward.
So the map ends up not containing any `probabilityDynamic` descriptor and the CutOnThreshold post filter fails.